### DISCUSSION TITLE
Can pypdf correctly extract text from two-column PDF documents?

### DISCUSSION THREAD

**Discussion by rybshik** [py-pdf/pypdf#2467] [Q&A]
Consider the following article: [https://arxiv.org/pdf/2106.13823.pdf](https://arxiv.org/pdf/2106.13823.pdf)

It is an academic paper formatted in two columns.

Can pypdf correctly extract text from PDF documents with a two-column format?

**Comment by MartinThoma** [comment]
PyPDF2 is deprecated and PyPDF3 / PyPDF4 are different projects. I've edited your question to be only about pypdf.

**Comment by MartinThoma** [comment]
Is the layout mode text extraction what you're looking for?

https://pypdf.readthedocs.io/en/stable/user/extract-text.html

**Reply by rybshik** [reply]
I want to extract text from a two-column PDF in the natural reading order: first down the left column, then down the right. 
Does PyPDF2 do this by default?

**Reply by pubpub-zz** [reply]
The standard `extract_text()` seems to work properly on your text. Have a try.

**Reply by pubpub-zz** [reply]
I mean `.extract_text(extraction_mode='plain'))`. .extract_text(extraction_mode='layout')) will put the columns on the same line
